### ISSUE TITLE
eventbridge rule name is fixed

### ISSUE THREAD

**Issue by roger-amorim-dv** [terraform-aws-modules/terraform-aws-eventbridge#49]
## Is your request related to a new offering from AWS?
Is this functionality available in the [AWS provider for Terraform](https://registry.terraform.io/providers/hashicorp/aws/latest/docs)? See [CHANGELOG.md](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md), too.
  - No 🛑: please wait to file a request until the functionality is avaialble in the AWS provider

## Is your request related to a problem? Please describe.
When executing the module using event scheduling with lambda and eventbridge we noticed that there is no parameterization for the name field of the eventbridge rule, it is created in a mocked way. This is a problem when we need to create more than one resource with the same characteristics.

## Describe the solution you'd like.

Changing this line: https://github.com/terraform-aws-modules/terraform-aws-eventbridge/blob/master/main.tf#L5. We can take this name in a parameterized way to generate the rule name.

## Describe alternatives you've considered.
Maybe there might be a way to generate the name randomly, but I still think it's not the best solution, given that we can pass these values ​​descriptively.

## Additional context
I believe that this improvement will positively impact whoever uses this module, so we can have a better and reusable module.

**Comment by loicvolle** [comment]
I also confirm that the name of the name of the rule is "crons-rule".

  rules = {
    crons = {
      name                = "Lambda-${var.app}-${var.ENV}"
      description         = "Trigger the Lambda ${var.app} check  and put metric into cloudwatch"
      schedule_expression = var.cloudwatch_cron
    }
  }

looks like the "crons" is taken and added "-rule" suffix instead of the variable name. The attribute is present in the provider and should be used

**Comment by baduker** [comment]
I can confirm that the issue still occurs. There's no way to change / update the rule name.

`module "eventbridge" {
  source = "terraform-aws-modules/eventbridge/aws"
  version = "1.14.0"

  create_bus   = false
  create_role  = false
  create_rules = true

  rules = {
    crons = {
      description         = "deafault"
      schedule_expression = "rate(1 day)"
    }
  }

  targets = {
    crons = [
      {
        arn   = module.lambda_function.lambda_function_arn
        input = jsonencode({ "job" : "crons" })
      }
    ]
  }
}`

After changes:

`
module "eventbridge" {
  ...
  rules = {
    crons = {
      description         = "My custom cron rule"
      schedule_expression = "rate(1 day)"
    }
  }

  targets = {
    crons = [
      {
        name  = "my-custom-cron-rule-name"
        arn   = module.lambda_function.lambda_function_arn
        input = jsonencode({ "job" : "crons" })
      }
    ]
  }
}
`

After `plan`, terraform picks up only the description change.

`Terraform will perform the following actions:

  # module.eventbridge.aws_cloudwatch_event_rule.this["crons"] will be updated in-place
  ~ resource "aws_cloudwatch_event_rule" "this" {
      ~ description         = "deafault" -> "My custom cron rule"
        id                  = "crons-rule"
        name                = "crons-rule"
        tags                = {
            "Name" = "crons-rule"
        }
        # (5 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
`

[Stackoverflow related question](https://stackoverflow.com/questions/72215332/change-eventbridge-cron-rule-name-in-terraform)

**Comment by baduker** [comment]
It's not an issue, but a feature. :)

Here's a relevant [Stackoverflow question](https://stackoverflow.com/questions/72215332/change-eventbridge-cron-rule-name-in-terraform).

TL;DR

You have to change this bit to name your rule:

```
  rules = {
    my-cron-rule = { <-- CHANGE HERE
      description     = "Trigger for a Lambda"
      event_pattern   = jsonencode({"source": ["aws.events"]})
      corn_expression = "cron(0 6 ? * THU *)"
    }
  }
```

**Comment by loicvolle** [comment]
Hi,

I made some tests and we can use some variables in the rule.


``` module "eventbridge" {
  source  = "terraform-aws-modules/eventbridge/aws"
  version = "1.14.0"
  create_bus = false
  rules = {
    "Lambda-${var.app}-${var.ENV}" = {
      name                = "Lambda-${var.app}-${var.ENV}"
      description         = "Trigger the Lambda ${var.app} check  and put metric into cloudwatch"
      schedule_expression = var.cloudwatch_cron
    }
  }
  targets = {
    "Lambda-${var.app}-${var.ENV}" = [
      {
        name = "${var.app}-${var.ENV}"
        arn  = module.lambda.lambda_function_arn
        input = jsonencode({
          endpoints        = var.ad_records.*.url
          metric_name      = var.metric_name
          metric_namespace = var.namespace
        })
      }
    ]
  }
}
